test/GeoPointTest.cpp

```diff
@@ -2,16 +2,121 @@
 //  Chair of Algorithms and Data Structures.
 //  Author: Christoph Ullinger <ullingec@informatik.uni-freiburg.de>
 
+#include <absl/strings/str_cat.h>
 #include <gtest/gtest.h>
 
-#include "global/GeoPoint.h"
+#include "global/Constants.h"
+#include "parser/GeoPoint.h"
 
 TEST(GeoPoint, GeoPoint) {
   GeoPoint g = GeoPoint(70.5, 130.2);
-  ASSERT_FLOAT_EQ(g.getLat(), 70.5);
-  ASSERT_FLOAT_EQ(g.getLng(), 130.2);
+
+  ASSERT_DOUBLE_EQ(g.getLat(), 70.5);
+  ASSERT_DOUBLE_EQ(g.getLng(), 130.2);
+
   ASSERT_THROW(GeoPoint(-99.5, 1.0), CoordinateOutOfRangeException);
   ASSERT_THROW(GeoPoint(99.5, 1.0), CoordinateOutOfRangeException);
   ASSERT_THROW(GeoPoint(9.5, -185.0), CoordinateOutOfRangeException);
   ASSERT_THROW(GeoPoint(9.5, 185.0), CoordinateOutOfRangeException);
+  ASSERT_NO_THROW(GeoPoint(90, 180));
+  ASSERT_NO_THROW(GeoPoint(0, 1.0));
+  ASSERT_NO_THROW(GeoPoint(1.0, -180.0));
+  ASSERT_NO_THROW(GeoPoint(0, 0));
+
+  g = GeoPoint(0, 0);
+  ASSERT_DOUBLE_EQ(g.getLat(), 0);
+  ASSERT_DOUBLE_EQ(g.getLng(), 0);
+  g = GeoPoint(90, 180);
+  ASSERT_DOUBLE_EQ(g.getLat(), 90);
+  ASSERT_DOUBLE_EQ(g.getLng(), 180);
+  g = GeoPoint(-90, -180);
+  ASSERT_DOUBLE_EQ(g.getLat(), -90);
+  ASSERT_DOUBLE_EQ(g.getLng(), -180);
+  g = GeoPoint(-90, 180);
+  ASSERT_DOUBLE_EQ(g.getLat(), -90);
+  ASSERT_DOUBLE_EQ(g.getLng(), 180);
+  g = GeoPoint(90, -180);
+  ASSERT_DOUBLE_EQ(g.getLat(), 90);
+  ASSERT_DOUBLE_EQ(g.getLng(), -180);
+  g = GeoPoint(0, 180);
+  ASSERT_DOUBLE_EQ(g.getLat(), 0);
+  ASSERT_DOUBLE_EQ(g.getLng(), 180);
+  g = GeoPoint(90, 0);
+  ASSERT_DOUBLE_EQ(g.getLat(), 90);
+  ASSERT_DOUBLE_EQ(g.getLng(), 0);
+}
+
+TEST(GeoPoint, string) {
+  GeoPoint g = GeoPoint(-70.5, -130.2);
+  ASSERT_EQ(g.toStringRepresentation(), "POINT(-130.200000 -70.500000)");
+
+  g = GeoPoint(90, 180);
+  ASSERT_EQ(g.toStringRepresentation(), "POINT(180.000000 90.000000)");
+
+  g = GeoPoint(0, 0);
+  ASSERT_EQ(g.toStringRepresentation(), "POINT(0.000000 0.000000)");
+
+  g = GeoPoint(-70.5, -130.2);
+  auto strpair = g.toStringAndType();
+  ASSERT_EQ(strpair.first, "POINT(-130.200000 -70.500000)");
+  ASSERT_EQ(strpair.second, GEO_WKT_LITERAL);
+}
+
+TEST(GeoPoint, bitRepresentation) {
+  GeoPoint g = GeoPoint(-70.5, -130.2);
+  constexpr double lat = ((-70.5 + 90) / (2 * 90)) * (1 << 30);
+  ASSERT_EQ(g.toBitRepresentation() >> 30, round(lat));
+  constexpr double lng = ((-130.2 + 180) / (2 * 180)) * (1 << 30);
+  ASSERT_EQ(g.toBitRepresentation() & ((1 << 30) - 1), round(lng));
+
+  constexpr size_t expect = (static_cast<size_t>(1) << 30) << 30 | (1 << 30);
```

Comment:
This leaks into the datatype bits, fix it, and add an assertion in the to"BitRepresentation, that those are in fact left untouched.